build/script.js

```diff
@@ -102,6 +104,34 @@ gulp.task('scripts-watch', createScriptsStream(false));
  * 'async'.
  */
 function createCompileTask(translation) {
+
+  let outputDir =
+      translation ? path.join(conf.paths.i18nProd, `/${translation.key}`) : conf.paths.prodTmp;
+
+  return (
+      (next) =>
+          gulp.src([
+                // Application source files.
+                path.join(conf.paths.frontendSrc, '**/*.js'),
+                // Partials generated by other tasks, e.g., Angular templates.
+                path.join(conf.paths.partials, '**/*.js'),
+                // Include base.js to enable some compiler functions, e.g., @export annotation
+                // handling and getMsg() translations.
+                path.join(
+                    conf.paths.bowerComponents, 'google-closure-library/closure/goog/base.js'),
+              ])
+              .pipe(closureCompiler(translation))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_DASHBOARD_VERSION', conf.deploy.version.release)))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_GIT_COMMIT', process.env.TRAVIS_COMMIT)))
```

Comment:
it seems that gulpIf works on destination files - not source files. I tried a lot (including reordering) but something like `**/appconfig_service.js` just did not work. So it seems I have to go through the hole file for replace. I could make the string more unique to make accidental replacement unlikely. 

I could use gulp-replace-task (https://www.npmjs.com/package/gulp-replace-task) instead of gulp-replace to refactor the to replace lines in a single function. 

Not sure. Maybe I should place the replace code into build.js WDYT?